flashinfer/fp4_quantization.py

```diff
@@ -62,9 +63,24 @@ def _pad_scale_factors(
         ).contiguous()
 
 
-def gen_fp4_quantization_sm100_module() -> JitSpec:
+def gen_fp4_quantization_module() -> JitSpec:
+    nvcc_flags = [
+        "-DENABLE_BF16",
+        "-DENABLE_FP8",
+        "-DENABLE_FP4",
+    ]
+
+    if get_device_arch() == "100a":
+        nvcc_flags += sm100a_nvcc_flags
+    elif get_device_arch() == "90a":
+        nvcc_flags += sm90a_nvcc_flags
+    else:
```

Comment:
Isn't this an issue to query the current GPU here? Wouldn't this break AOT build where we want to build independently of even having a GPU available?